tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -345,11 +345,27 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
 
     // Class definition
     if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
-      val ctor = classTreeCache.constructor.getOrElseUpdate(
-          classEmitter.genConstructor(linkedClass)(classCache))
+      val (linkedInlineableInit, linkedMemberMethods) =
+        classEmitter.extractInlineableInit(linkedClass)(classCache)
+
+      // JS constructor
+      val ctor = {
+        /* The constructor depends both on the class version, and the version
+         * of the inlineable init. When there is no inlineable init, we use ""
+         * as its version.
+         */
+        val ctorCache = classCache.getConstructorCache()
+        val optInlineableInitVersion =
+          linkedInlineableInit.fold[Option[String]](Some(""))(_.version)
```

Comment:
I'd rather have that, yes. Sorry for not spotting it earlier. The problem with implicit guarantees is that they are implicit :)